homeassistant/components/kef/media_player.py

```diff
@@ -0,0 +1,270 @@
+"""Platform for the KEF Wireless Speakers."""
+
+from datetime import timedelta
+import logging
+
+from aiokef.aiokef import AsyncKefSpeaker
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    PLATFORM_SCHEMA,
+    SUPPORT_SELECT_SOURCE,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+    SUPPORT_VOLUME_MUTE,
+    SUPPORT_VOLUME_SET,
+    SUPPORT_VOLUME_STEP,
+    MediaPlayerDevice,
+)
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PORT,
+    CONF_TYPE,
+    STATE_OFF,
+    STATE_ON,
+    STATE_UNKNOWN,
+)
+from homeassistant.helpers import config_validation as cv
+
+_CONFIGURING = {}
+_LOGGER = logging.getLogger(__name__)
+
+
+DEFAULT_NAME = "KEF"
+DEFAULT_PORT = 50001
+DEFAULT_MAX_VOLUME = 0.5
+DEFAULT_VOLUME_STEP = 0.05
+DEFAULT_INVERSE_SPEAKER_MODE = False
+
+DOMAIN = "kef"
+
+SCAN_INTERVAL = timedelta(seconds=30)
+PARALLEL_UPDATES = 0
+
+SOURCES = {"LSX": ["Wifi", "Bluetooth", "Aux", "Opt"]}
+SOURCES["LS50"] = SOURCES["LSX"] + ["Usb"]
+
+SUPPORT_KEF = (
+    SUPPORT_VOLUME_SET
+    | SUPPORT_VOLUME_STEP
+    | SUPPORT_VOLUME_MUTE
+    | SUPPORT_SELECT_SOURCE
+    | SUPPORT_TURN_OFF
+    | SUPPORT_TURN_ON
+)
+
+CONF_MAX_VOLUME = "maximum_volume"
+CONF_VOLUME_STEP = "volume_step"
+CONF_INVERSE_SPEAKER_MODE = "inverse_speaker_mode"
+CONF_STANDBY_TIME = "standby_time"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_TYPE): vol.In(["LS50", "LSX"]),
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_MAX_VOLUME, default=DEFAULT_MAX_VOLUME): cv.small_float,
+        vol.Optional(CONF_VOLUME_STEP, default=DEFAULT_VOLUME_STEP): cv.small_float,
+        vol.Optional(
+            CONF_INVERSE_SPEAKER_MODE, default=DEFAULT_INVERSE_SPEAKER_MODE
+        ): cv.boolean,
+        vol.Optional(CONF_STANDBY_TIME): vol.In([20, 60]),
+    }
+)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the KEF platform."""
+    if DOMAIN not in hass.data:
+        hass.data[DOMAIN] = {}
+
+    host = config[CONF_HOST]
+    speaker_type = config[CONF_TYPE]
+    port = config.get(CONF_PORT)
```

Comment:
Please use `dict[key]` for required config keys and keys with default config schema values.

Reply:
I am already doing that.

`port` is optional and only `host` and `speaker_type` are required.

Reply:
Port has a default value.

Reply:
do you mean that it should be `port = config.get(CONF_PORT, DEFAULT_PORT)`?

Reply:
No. It should be:
```py
port = config[CONF_PORT]
```
The config schema makes sure there's a default value for this key.

Reply:
Oh I see, fixed it!